File: Fibo_dp.py
import numpy as np

#initializing our recursive memo function
def Fibonacci(n, dp): 
    #Exit Condition
    if n<2:
        dp[n] = 1
        return 1
    #else
    if(dp[n]!= 0): 
        return dp[n]
        #else    
    dp[n] = Fibonacci(n-1, dp)+Fibonacci(n-2, dp) 
    return dp[n]


#initializing our memo table
dp = np.zeros((50), dtype = "int32")

#Reading our n-th index needed
N = int(input()) -1

print(Fibonacci(N, dp)) 

File: Fibo_dp.cpp
#include <bits/stdc++.h> 
using namespace std; 


//Global Memoization table so we can access it inside our function
int dp[50]; 


//Caculating the n-th Fibonacci Number using our memo-recursive function
int fibonacci(int n){
    //Exit condition
    if(n==1 || n==2){
        dp[n] = 1 ; 
        return 1 ; 
    }
    //else
        if(dp[n]!=0) return dp[n]; 
        //else
        dp[n] = fibonacci(n-1) + fibonacci(n-2) ; 
        return dp[n] ; 
}

int main(){

    // Initializing our dp table
    for (int i = 0; i < 50; i++)
    {
        dp[i] = 0 ; 
    }
    
    int N; 
    cin>>N ; 
    
    cout<<fibonacci(N) ; 
    return 0;
}